### ISSUE TITLE
Chapter 2 " optimizer.zero_grad()"

### ISSUE THREAD

**Issue by RoastedKernel** [EdwardRaff/Inside-Deep-Learning#20]
In the beginning of chapter 2 
2.1.3  The training loop

you used the following steps in you train simple nn function. 
            optimizer.zero_grad()
            loss.backward()
            optimizer.step()

but in the end of the chapter for def run_epoch function you used a different order:

2.4.2  Training and testing passes

if model.training: 
   loss.backward() 
   optimizer.step() 
   optimizer.zero_grad()
   
  is there any reason for switching the order?  i thought we had to zero the gradients firs thing at every epoch.

**Comment by RoastedKernel** [comment]
Nevermind. As long as the gradients are zeroed by the next batch. it should be fine. Great book by the way.
